torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -49,5 +49,18 @@ Tensor LinearImpl::forward(const Tensor& input) {
   AT_ASSERT(!options.with_bias() || bias.defined());
   return torch::linear(input, weight, bias);
 }
+
+// ============================================================================
+
+void FlattenImpl::reset() {}
+
+void FlattenImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::Flatten()";
+}
+
+Tensor FlattenImpl::forward(const Tensor& input) {
+  return input.flatten();
```

Comment:
In Python version, `torch.nn.Flatten` takes `start_dim` and `end_dim` as optional constructor arguments, and we'd need to do the same for the C++ version.

Reply:
We would also need to add tests for the optional constructor arguments `start_dim` and `end_dim` as well. Thanks a lot for your help!